### PR TITLE
Fix the issue where the agent interface cannot call the context

### PR DESCRIPTION
**Pull request by kunkeji**
### What problem does this PR solve?

Fix the context of the agent interface call to the context during web testing, and change it to the context record of user chat
Remove duplicate messages and add them, which can be viewed in the messages section of database api_4_comversation

### Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by JinHai-CN** [comment]
@kunkeji Would you please use English to write the code comment? Thank you for your contribution!

**Comment by kunkeji** [comment]
@JinHai-CN I will definitely do it next time

**Comment by JinHai-CN** [comment]
Just modify the code and push one more commit to update the PR.

**Comment by JinHai-CN** [comment]
git pull -> update code -> commit -> git push

File: api/apps/sdk/session.py [hunk 1]
```diff
@@ -308,7 +308,7 @@ def rename_field(ans):
             if 'docnm_kwd' in chunk_i:
                 chunk_i['doc_name'] = chunk_i['docnm_kwd']
                 chunk_i.pop('docnm_kwd')
-    conv.message.append(msg[-1])
+    # conv.message.append(msg[-1])   Remove duplicate additions
```
Comment (KevinHuSh) [inline api/apps/sdk/session.py]:
```suggestion
```

File: api/apps/sdk/session.py [hunk 2]
```diff
@@ -317,9 +316,13 @@ def rename_field(ans):
 
     final_ans = {"reference": [], "content": ""}
 
-    canvas.messages.append(msg[-1])
+    canvas.messages = conv.message      # Correct the context to interface context
```
Comment (KevinHuSh) [inline api/apps/sdk/session.py]:
```suggestion
```

File: api/apps/sdk/session.py [hunk 3]
```diff
@@ -317,9 +316,12 @@ def rename_field(ans):
 
     final_ans = {"reference": [], "content": ""}
 
-    canvas.messages.append(msg[-1])
     canvas.add_user_input(msg[-1]["content"])
 
+    # Historical record analysis
```
Comment (KevinHuSh) [inline api/apps/sdk/session.py]:
```suggestion
```

File: api/apps/sdk/session.py [hunk 4]
```diff
@@ -317,9 +316,11 @@ def rename_field(ans):
 
     final_ans = {"reference": [], "content": ""}
 
-    canvas.messages.append(msg[-1])
     canvas.add_user_input(msg[-1]["content"])
 
+    for m in conv.message:
```
Comment (KevinHuSh) [inline api/apps/sdk/session.py]:
```suggestion
```

File: api/apps/sdk/session.py [hunk 5]
```diff
@@ -317,9 +316,10 @@ def rename_field(ans):
 
     final_ans = {"reference": [], "content": ""}
 
-    canvas.messages.append(msg[-1])
     canvas.add_user_input(msg[-1]["content"])
 
+        canvas.history.append((m["role"], m["content"]))
```
Comment (KevinHuSh) [inline api/apps/sdk/session.py:321]:
```suggestion
```

File: api/apps/sdk/session.py [hunk 6]
```diff
@@ -317,9 +316,9 @@ def rename_field(ans):
 
     final_ans = {"reference": [], "content": ""}
 
-    canvas.messages.append(msg[-1])
     canvas.add_user_input(msg[-1]["content"])
 
+
```
Comment (KevinHuSh) [inline api/apps/sdk/session.py:321]:
```suggestion
```

**Pull request merged**